src/redux/missions/missionsSlice.js

```diff
@@ -0,0 +1,75 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+// async thunk for fetching missions
+export const fetchMissions = createAsyncThunk('missions/fetchMissions', async () => {
+  try {
+    const response = await axios.get('https://api.spacexdata.com/v3/missions');
+    return response.data;
+  } catch (error) {
+    throw new Error('Failed to fetch missions.');
+  }
+});
+
+export const joinMission = createAsyncThunk('missions/joinMission', async (missionName, { getState }) => {
+  const state = getState();
+  const mission = state.missions.missions.find((mission) => mission.mission_name === missionName);
+  if (mission) {
+    return missionName;
+  }
+  throw new Error('Mission not found.');
+});
+
+export const leaveMission = createAsyncThunk('missions/leaveMission', async (missionName, { getState }) => {
+  const state = getState();
+  const mission = state.missions.missions.find((mission) => mission.mission_name === missionName);
+  if (mission) {
+    return missionName;
+  }
+  throw new Error('Mission not found.');
+});
```

Comment:
- Optional.
`createAsyncThunk` is specifically designed for handling async actions like API calls. In the case of `joinMission` and `leaveMission`, which are simple actions, it is not necessary to use `createAsyncThunk`. Instead, you can define them directly within the reducers as regular action creators. By doing so, you will eliminate the overhead of `createAsyncThunk`, which is more suitable for handling asynchronous actions. This will simplify the code and make it easier to reason about.